tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -4536,8 +4540,15 @@ SpirvEmitter::incDecRWACSBufferCounter(const CXXMemberCallExpr *expr,
     return nullptr;
   }
 
+  llvm::SmallVector<SpirvInstruction *, 2> indexes;
+  if(const auto *arraySubscriptExpr = dyn_cast<ArraySubscriptExpr>(object)) {
+    //TODO: Handle multi-dimensional arrays.
```

Comment:
Is handling multi-dimensional arrays more complex than doing:

```cpp
auto head = object;
while (const auto *arraySubscriptExpr = dyn_cast<ArraySubscriptExpr>(head)) {
   indexes.push_back(doExpr(arraySubscriptExpr->getIdx());
   head = arraySubscriptExpr;
}
```

?

Reply:
I don't know how complex it is. I have not looked at an example yet. We may need to reverse the subscripts. I don't want to write the code until I have a test, and I want to keep the PRs focused.

Reply:
Fine by me then. Would be nice to have an issue open referencing this issue in case someones hits this issue.

Reply:
Sorry, I had updated the comments, but the changes did not get pushed. Today I created an over arching issue, and I link to it now.